package.json

```diff
@@ -7,42 +7,43 @@
     "doc": "docs"
   },
   "devDependencies": {
-    "@babel/cli": "^7.12.10",
-    "@babel/core": "^7.12.10",
-    "@babel/plugin-transform-runtime": "^7.12.10",
-    "@babel/plugin-transform-typescript": "^7.12.1",
-    "@babel/preset-env": "^7.12.10",
-    "@babel/preset-react": "^7.12.10",
-    "@babel/types": "^7.12.10",
-    "@types/jest": "^26.0.18",
-    "@types/react": "^16.9.23",
-    "@types/react-dom": "^16.9.5",
+    "@babel/cli": "^7.20.7",
+    "@babel/core": "^7.20.12",
+    "@babel/plugin-transform-runtime": "^7.19.6",
+    "@babel/plugin-transform-typescript": "^7.20.13",
+    "@babel/preset-env": "^7.20.2",
+    "@babel/preset-react": "^7.18.6",
+    "@babel/types": "^7.20.7",
+    "@types/jest": "^28.1.3",
+    "@types/react": "^16.14.35",
+    "@types/react-dom": "^16.9.17",
     "@types/turbolinks": "^5.2.0",
-    "@types/webpack-env": "^1.17.0",
-    "@typescript-eslint/eslint-plugin": "^4.10.0",
-    "@typescript-eslint/parser": "^4.10.0",
+    "@types/webpack-env": "^1.18.0",
+    "@typescript-eslint/eslint-plugin": "^4.33.0",
+    "@typescript-eslint/parser": "^4.33.0",
     "babelify": "^10.0.0",
     "blue-tape": "^1.0.0",
-    "create-react-class": "^15.6.0",
-    "eslint": "^7.15.0",
+    "create-react-class": "^15.7.0",
+    "eslint": "^7.32.0",
     "eslint-config-prettier": "^7.0.0",
     "eslint-config-shakacode": "^16.0.1",
-    "eslint-plugin-import": "^2.22.1",
-    "eslint-plugin-jsx-a11y": "^6.4.1",
-    "eslint-plugin-prettier": "^3.3.0",
-    "eslint-plugin-react": "^7.21.5",
-    "jest": "^26.6.3",
+    "eslint-plugin-import": "^2.27.5",
+    "eslint-plugin-jsx-a11y": "^6.7.1",
+    "eslint-plugin-prettier": "^3.4.1",
+    "eslint-plugin-react": "^7.32.1",
+    "jest": "^28.1.3",
+    "jest-environment-jsdom": "^28.1.3",
     "jsdom": "^16.4.0",
     "nps": "^5.9.3",
-    "prettier": "^2.2.1",
+    "prettier": "^2.8.3",
     "prettier-eslint-cli": "^5.0.0",
-    "prop-types": "^15.5.10",
-    "react": "^16.5.2",
-    "react-dom": "^16.5.2",
+    "prop-types": "^15.8.1",
+    "react": "^16.14.0",
```

Comment:
Why is react not upgraded to 17 at least?

Reply:
There was no security issue with React. For such dependencies, I only pushed all minor versions. I thought it might be a development decision to use which version of React.